lib/SILOptimizer/Mandatory/DiagnoseUnreachable.cpp

```diff
@@ -568,7 +568,8 @@ static bool simplifyBlocksWithCallsToNoReturn(SILBasicBlock &BB,
   // much smaller and easier to read, but otherwise has no effect.
   SILBuilder B(&BB);
   B.setCurrentDebugScope(Scope);
-  B.createUnreachable(ArtificialUnreachableLocation());
+  // Sequencial linetables are helpful for WinDbg.
```

Comment:
I vaguely recall that ArtificialUnreachableLocation had special semantics for Diagnostics. Also.. how does this confuse WinDbg?

Reply:
This is to fix the linetables in the ``/`` instruction. The ``NoReturnCall`` and the ``unreachable`` had different debug locations which broke up the ``/`` instruction into multiple debug lines. I don't see why it is necessary to make this ``unreachable`` have an artificial location.

Reply:
According to the comment in SILLocation
```
/// Differentiates an unreachable instruction, which is generated by
/// DCE, from an unreachable instruction in user code (output of SILGen).
/// Allowed on an unreachable instruction.
```
ArtificialUnreachableLocation has a very specific meaning. I haven't done the detective work to figure out where this difference matters, but I don't want to change SIL generation without having made sure that it's safe to remove it first.

Reply:
Earlier I had another fix for this. In IRGenDebugInfo I checked for every instance of ``ArtificialUnreachableLocation`` and set its locations to be the same as the previous instruction. This would only need to be done when generating CodeView and would not affect the SIL.

Reply:
I would feel more comfortable doing it this way, thanks!